aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -537,8 +532,50 @@ impl AptosVMImpl {
                 ]),
                 &mut UnmeteredGasMeter,
             )
-            .map(|_return_vals| ())
-            .map_err(expect_no_verification_errors)
+        } else {
+            // Gas payer tx
+            let gas_payer = *txn_data.secondary_signers.last().ok_or_else(|| {
+                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                    .finish(Location::Undefined)
+            })?;
+            session.execute_function_bypass_visibility(
+                &transaction_validation.module_id(),
+                &transaction_validation.user_epilogue_gas_payer_name,
```

Comment:
Can we run the same epilogue function but with gas payer set to sender? This would allow having a single epilogue function instead of 2

Reply:
No replay and forking requires to call the old epilogue function

Reply:
it needs to be different if i understand correctly, it bumps the sequence number of sender but charges gas of the payer so that becomes two identity instead of one

Reply:
May be just to be safe but breaking older binaries with new framework compat is not uncommon. Every gas schedule upgrade breaks compat with older binaries

Reply:
But the issue is not breaking newer binaries with older framework. Ie. new binaries must continue to call epilogue function without a gas_payer, because that's the only one existing in old framework. 

Anyway given that the old epilogue is implemented in terms of the newer epilogue, I don't think it's a big deal.

Reply:
Yeah makes sense